### PR TITLE
scratch: use winrar to extract nsis sfx

### LINKED ISSUE #7338: Scratch install don't work anymore (last update)
## Bug Report

**Scratch**

### Current Behaviour
image

### Expected Behaviour

Correct installation

### Additional context/output

Tested the downloaded version. It's not an autoextractible .exe like previously. It's a classic .exe which is not extractible with 7zip.

### System details

**Windows version:** [e.g. 7, 8, 10]

**OS architecture:** [e.g. 32bit, 64bit]

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]

**Additional software:** [(optional) e.g. ConEmu, Git]

### PR DESCRIPTION
**Pull request by phanirithvij**
* added winrar as a depenceny
* fixes #7338
* This is a workaround

### CODE REVIEW

**Comment by issaclin32** [comment]
My concern is that `"depends": "winrar"` can bring some confusing situations for end-users.

Most people have their WinRAR installed through other means. If we silently install another WinRAR without asking the user, people may complain about it.

I tend to add a new package named `winrar-helper`, which is almost indentical to current `winrar`, but does not add any bin/shortcuts. We can always keep `winrar-helper` updated, so it is better than detecting WinRAR installations in the script.

**Comment by phanirithvij** [comment]
Where is `winrar-helper` located I don't see it in this bucket, is it yet to be created? If so I will make that fix in a new PR.

**Comment by issaclin32** [comment]
> Where is `winrar-helper` located I don't see it in this bucket, is it yet to be created? If so I will make that fix in a new PR.

It will be created in another PR. However I cannot decide this on my own, so I will wait for other maintainers' opinions in that PR. It will be merged in a few days. Please give us a little bit of time.

**Comment by phanirithvij** [comment]
I understand.
I was thinking as an alternative, we can download winrar to a temp directory in this scratch.json manifest `pre_install` step itself and use it to extract this nsis sfx.
But I agree that a helper in the bucket is more useful for keeping winrar up to date and also for other packages which might require winrar.

File: bucket/scratch.json [hunk 1]
```diff
@@ -6,11 +6,15 @@
         "identifier": "Freeware",
         "url": "https://scratch.mit.edu/info/faq"
     },
-    "url": "https://downloads.scratch.mit.edu/desktop/Scratch%203.27.0%20Setup.exe#/dl.7z",
+    "depends": "winrar",
```
Comment (phanirithvij) [inline bucket/scratch.json:9]:
```suggestion
    "depends": "winrar-helper",
```

File: bucket/scratch.json [hunk 2]
```diff
@@ -6,11 +6,15 @@
...
+    "url": "https://downloads.scratch.mit.edu/desktop/Scratch%203.27.0%20Setup.exe#/setup.exe",
     "hash": "343135e697db089a7ce6e8736950c4693b418d7cbc41151e33a45cac4ecc89db",
     "pre_install": [
-        "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-32.7z\" \"$dir\"",
-        "Remove-Item \"$dir\\`$*\", \"$dir\\Uninstall*\" -Recurse"
+        "$Args = @\"",
+        "x -IBCK \"$dir\\setup.exe\" \"$dir\"",
+        "\"@",
+        "Start-Process -Wait winrar.exe -NoNewWindow -ArgumentList $Args",
```
Comment (phanirithvij) [inline bucket/scratch.json:16]:
```suggestion
        "Start-Process -Wait \"$(appdir winrar-helper $global)\\current\\winrar.exe\" -NoNewWindow -ArgumentList $Args",
```

**Review by phanirithvij — COMMENTED** [review]
Once #7390 is reviewed these changes can be made

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**